src/yarn.ts

```diff
@@ -1355,24 +1355,34 @@ const completionSpec: Fig.Spec = {
         const version = await executeShellCommand("yarn --version");
         const isYarnV1 = version.startsWith("1.");
 
-        // Only use info in yarn workspaces info 1.X.X
-        const versionedCommand = isYarnV1 ? "info" : "list --json";
+        const getWorkspacesDefinitionsV1 = async () => {
+          const out = await executeShellCommand(`yarn workspaces info`);
 
-        try {
-          const out = await executeShellCommand(
-            `yarn workspaces ${versionedCommand}`
-          );
+          const startJson = out.indexOf("{");
+          const endJson = out.lastIndexOf("}");
+
+          return Object.entries(
+            JSON.parse(out.slice(startJson, endJson + 1)) as Record<
+              string,
+              { location: string }
+            >
+          ).map(([name, { location }]) => ({
+            name,
+            location,
+          }));
+        };
 
+        const getWorkspacesDefinitionsVOther = async () => {
```

Comment:
Let's leave this comment for future devs :)
```suggestion
        // For yarn >= 2.0.0
        const getWorkspacesDefinitionsVOther = async () => {
```